### DISCUSSION TITLE
Dot-repeat in visual mode

### DISCUSSION THREAD

**Discussion by tvsfx** [kylechui/nvim-surround#300] [Design]
I usually use `nvim-surround` in visual mode, but noticed that dot-repeat does not work there. This is confirmed in the [Dev Wiki](https://github.com/kylechui/nvim-surround/wiki/For-Developers).  Is there any particular reason why this might be the case? Is it because the motions used in visual mode are more difficult to retrieve somehow, which prevents your trick (https://gist.github.com/kylechui/a5c1258cd2d86755f97b10fc921315c3) from working? Is there some workaround to make it work? Thanks in advance!

**Comment by kylechui** [comment]
It's because once you select something in visual mode, there is no concept of the motion used to create the selection (since you might have used many motions many times, e.g. `iwwll`). Thus we only really have start/end coordinates available, which isn't particularly helpful for dot-repeating.

**Reply by tvsfx** [reply]
Yes, that's more or less what I expected. However, dot-repeat for e.g. `c`hange or `d`elete in visual mode _does_ work, so it looks like nvim manages to track and replay all of the motions that the user performed before a change/delete. Is this a limitation of the way you implement dot-repeat?

**Reply by kylechui** [reply]
Based on testing on my own machine, it doesn't work. It seems to change/delete the same number of characters, but does not respect the motion that generated that selection.

**Reply by tvsfx** [reply]
You're right, I thought I tested for that, but I must've coincidentally picked words of similar length. My bad.

So does that mean you could theoretically still dot-surround words of the same size? That could be useful to surround the same selection multiple times, outside-in.
In any case, I guess I'll switch to surrounding without visual mode then, that seems more convenient. Thanks!

**Reply by kylechui** [reply]
Current implementation doesn't support dot-repeating in visual mode at all; I imagine 99.9% of use cases revolve around surround semantic text objects, rather than exactly `k` characters or something to that effect.
